src/components/FlexibleForm/ControlDecorators/withBooleanActions/withBooleanActions.js

```diff
@@ -0,0 +1,112 @@
+import React, {
+  memo,
+  useRef,
+  useState,
+  useLayoutEffect,
+} from 'react';
+import { PropTypes } from 'prop-types';
+import { FormattedMessage, injectIntl } from 'react-intl';
+
+import { get } from 'lodash';
+
+import { Select } from '@folio/stripes/components';
+
+import {
+  compose,
+  isFormattedMessage,
+  isTranslationId,
+  ENTITY_KEYS,
+  FORMS_SETTINGS,
+} from '../../../../utils';
+
+import styles from './withBooleanActions.css';
+
+const withBooleanActionsControl = props => {
+  const {
+    intl,
+    id,
+    input,
+    WrappedComponent,
+    wrapperLabel,
+    label,
+    ...rest
+  } = props;
+
+  const actions = get(FORMS_SETTINGS, [ENTITY_KEYS.MAPPING_PROFILES, 'DECORATORS', 'BOOLEAN_ACTIONS'], []);
+  const dataOptions = actions.map(action => ({
+    value: action.value,
+    label: intl.formatMessage({id: action.label}),
+  }));
+
+  const [currentValue, setCurrentValue] = useState(input?.value || '');
+  const [wrapperValue, setWrapperValue] = useState(null);
+
+  useLayoutEffect(() => {
+    let newValue;
+
+    if (wrapperValue) {
+      newValue = wrapperValue;
+    }
+
+    setCurrentValue(newValue);
+    input.onChange(newValue);
+  }, [wrapperValue]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
You don't need to use useLayoutEffect() here, because you don't retrieve data from the server. See how it is done in the `withDatePicker`

Reply:
fixed